src/cascadia/TerminalApp/TerminalPage.cpp

```diff
@@ -3010,9 +3052,24 @@ namespace winrt::TerminalApp::implementation
             co_return;
         }
 
+        auto pkgList = co_await _FindPackageAsync(args.MissingCommand());
```

Comment:
nit: can we initialize `pkgList` to `nullptr` here and then just increase the retry count by 1? That way we aren't repeating the `_FindPackageAsync` call

Reply:
I'd also consider moving the retry logic inside the `_FindPackageAsync` call. I feel like either retrying is always necessary for robustness or it's never necessary because it's already always robust. Put differently, `_FindPackageAsync` is "fallible call + additional logic" and so to me it shouldn't be `(fallible + additional) * 3` but rather `fallible * 3 + additional`.